src/util/metadata/side_metadata/global.rs

```diff
@@ -16,27 +16,125 @@ use std::sync::atomic::{AtomicU16, AtomicU32, AtomicU8, AtomicUsize, Ordering};
 #[derive(Clone, Copy, PartialEq, Eq, Hash)]
 pub struct SideMetadataSpec {
     pub is_global: bool,
-    pub offset: usize,
+    pub offset: SideMetadataOffset,
     /// Number of bits needed per region. E.g. 0 = 1 bit, 1 = 2 bit.
     pub log_num_of_bits: usize,
     /// Number of bytes of the region. E.g. 3 = 8 bytes, 12 = 4096 bytes (page).
     pub log_min_obj_size: usize,
 }
 
+impl SideMetadataSpec {
+    /// Is offset for this spec Address? (contiguous side metadata for 64 bits, and global specs in 32 bits)
+    #[inline(always)]
+    pub const fn is_addr_offset(&self) -> bool {
+        self.is_global || cfg!(target_pointer_width = "64")
+    }
+    /// If offset for this spec relative? (chunked side metadata for local specs in 32 bits)
+    #[inline(always)]
+    pub const fn is_rel_offset(&self) -> bool {
+        !self.is_addr_offset()
+    }
+
+    #[inline(always)]
+    pub const fn get_addr_offset(&self) -> Address {
```

Comment:
Not a strong opinion, but probably this should be `get_addr` or `get_absolute_offset` rather than `get_addr_offset`.

Reply:
Sounds good. I will change this.